client/lib/purchases/index.js

```diff
@@ -110,6 +110,9 @@ function getDisplayName( purchase ) {
 	if ( jetpackProductsDisplayNames[ purchase.productSlug ] ) {
 		return jetpackProductsDisplayNames[ purchase.productSlug ];
 	}
+	if ( isDomainRegistration( purchase ) || isDomainMapping( purchase ) ) {
```

Comment:
This appears to be a remnent of a previous attempt; it's been removed and in `renderDomainMappingDialogText()`, it's calling `purchase.productName` instead, in order to show "Domain Connection" (or whatever this may be called in the future).